### PR TITLE
Modify examples how about to correct the local time in clients 

### PR DESCRIPTION
**Pull request by chengwwYunify**
Sign in UploadManager is a different way from other requests. (There are many requests requested in this class. When a request need sign with server, the callback calls its method. )

So there are some changes in classes UploadManager and UploadManagerCallback.

       -  Not all of users need sign with server. We make methods "String onSignature(String strToSign)" and "String onAccessKey()" do not auto override.
        - To support java 1.6 as well, we make interfaceUploadManagerCallback become an abstract class .

 In summary, interface UploadManagerCallback is a abstract class now.

### CODE REVIEW

File: docs/example/sign_with_server.md [hunk 1]
```diff
@@ -9,37 +29,41 @@ Take uploading object for example:
...
+    // Step 4: serverAuthorization. Get response from server. We just sign in local here.
+    String serverAuthorization = QSSignatureUtil.generateSignature("secretKey",
+        strToSignature);
 
-        // Step 5: set the signature to the request.
-        reqHandler.setSignature("accessKey", serverAuthorization);
-        // Step 6: send request. Async requests use the method sendAsync(), sync requests use the method send().
-        reqHandler.sendAsync();
+    // Step 5: set the signature to the request.
+    reqHandler.setSignature("accessKey", serverAuthorization);
```
Comment (frostyplanet) [inline docs/example/sign_with_server.md]: 这里Step 5 可以加上设置 Date的提示，

File: docs/example/sign_with_server_zh.md [hunk 2]
```diff
@@ -1,45 +1,66 @@
...
 
-        // 第五步：将计算的签名设置到 request 中
-        reqHandler.setSignature("accessKey", serverAuthorization);
-        // 第六步：发送请求。异步请求使用 sendAsync() 方法。同步请求使用 send() 方法。
-        reqHandler.sendAsync();
+    // 第四步：serverAuthorization。server 端处理返回信息，服务端参考如下代码：
+    // String serverAuthorization = QSSignatureUtil.generateSignature("您的 secretKey", strToSignature);
+    String serverAuthorization = "您从服务端获取到的签名字符串";
+    // 第五步：将计算的签名设置到 request 中
+    reqHandler.setSignature("您的 accessKey", serverAuthorization);
```
Comment (frostyplanet) [inline docs/example/sign_with_server_zh.md]: 这里同上
